src/Context/FixtureContext.php

```diff
@@ -493,7 +493,14 @@ public function stepUpdateRecordState($type, $id, $state)
                 Versioned::set_reading_mode($oldMode);
                 break;
             case 'deleted':
-                $obj->delete();
+                if ($obj->hasExtension(Versioned::class)) {
+                    $obj->suppressDeletedVersion(fn () => $obj->delete());
+                } else {
+                    $obj->delete();
+                }
+                break;
+            case 'archived':
+                $obj->doArchive();
```

Comment:
Note I haven't added a check here to see if versioned is installed or if the record has the versioned extension, which is consistent with other functionality already in this same method. Archiving can only be done on versioned records, after all, so saying "this record is archived" in a behat test against a non-versioned record would be silly.